src/plugins/hetero/plugin.cpp

```diff
@@ -179,30 +185,44 @@ Parameter Engine::GetMetric(const std::string& name, const std::map<std::string,
-    } else if (ov::device::architecture == name) {
-        auto tconfig = mergeConfigs(_config, options);
-        std::string targetFallback = GetTargetFallback(tconfig);
-        return decltype(ov::device::architecture)::value_type{DeviceArchitecture(targetFallback)};
     } else {
         IE_THROW() << "Unsupported metric key: " << name;
     }
 }
-std::string Engine::DeviceArchitecture(const std::string& targetFallback) const {
+
+std::string Engine::DeviceCachingProperties(const std::string& targetFallback) const {
     auto fallbackDevices = ov::DeviceIDParser::get_hetero_devices(targetFallback);
-    std::string resArch;
+    // Vector of caching configs for devices
+    std::vector<ov::AnyMap> result = {};
     for (const auto& device : fallbackDevices) {
         ov::DeviceIDParser parser(device);
-
-        auto supportedMetricKeys = GetCore()
-                                       ->GetMetric(parser.get_device_name(), METRIC_KEY(SUPPORTED_METRICS))
-                                       .as<std::vector<std::string>>();
-        auto it = std::find(supportedMetricKeys.begin(), supportedMetricKeys.end(), METRIC_KEY(DEVICE_ARCHITECTURE));
-        auto arch = (it != supportedMetricKeys.end())
-                        ? GetCore()->GetMetric(device, METRIC_KEY(DEVICE_ARCHITECTURE)).as<std::string>()
-                        : parser.get_device_name();
-        resArch += " " + arch;
+        ov::AnyMap properties = {};
+        // Use name without id
+        auto device_name = parser.get_device_name();
+        auto supported_properties =
+            GetCore()->GetMetric(device, ov::supported_properties.name()).as<std::vector<ov::PropertyName>>();
+        if (std::find(supported_properties.begin(), supported_properties.end(), ov::caching_properties.name()) !=
+            supported_properties.end()) {
+            auto caching_properties =
+                GetCore()->GetMetric(device, ov::caching_properties.name()).as<std::vector<ov::PropertyName>>();
+            if (caching_properties.size()) {
+                for (auto& property_name : caching_properties) {
+                    properties[property_name] = GetCore()->GetMetric(device, property_name);
+                }
+            }
+            // If caching properties are not supported by device, try to add at least device architecture
+        } else if (std::find(supported_properties.begin(),
+                             supported_properties.end(),
+                             ov::device::architecture.name()) != supported_properties.end()) {
+            auto device_architecture = GetCore()->GetMetric(device, ov::device::architecture.name());
+            properties = ov::AnyMap{{ov::device::architecture.name(), device_architecture}};
+            // Device architecture is not supported, add device name as achitecture
+        } else {
+            properties = ov::AnyMap{{ov::device::architecture.name(), device_name}};
```

Comment:
It doesn't guarantee anything. During second launch device ids can be changed. And Device.1 can became Device.2. Both scenarious are not reliable and both are hypothetical. 

But actually I don't know plugins which supports export, can have more then 1 device and don't support caching properties/architecture. I think it is mostly needed for devices where device can have 1 type only, but device id can be different. Moreover the same code is located in core part, where we can have exactly the same situation https://github.com/openvinotoolkit/openvino/blob/a20b3631fb5d253bde84d82087cb11b3dee78c59/src/inference/src/dev/core_impl.cpp#L1257 

I propose to keep backward compatible bahavior in this PR. Anyway if we will faced with such plugin recommendation will be support caching properties.

Reply:
I suppose this `else` branch is valid only if device support import_export at all.
Otherwise, we save subgraph to IR and this piece of model is totally device independent (no needs to encode it somehow)